src/Database/Type/DateTimeType.php

```diff
@@ -210,7 +210,6 @@ public function toPHP(mixed $value, Driver $driver): DateTime|DateTimeImmutable|
 
         if (
             !$this->keepDatabaseTimezone
-            && $instance->getTimezone()
```

Comment:
`getTimezone()` could return `false` if the instance is just `DateTimeImmutable`, couldn't it?

Reply:
Maybe this was true in the past but not anymore
https://onlinephp.io/c/5f33d

Reply:
Mhmm ok it returns false "on failure" according to the docs. I wonder what kind of "failure" this is...

Reply:
You gotta love the PHP docs, they're always very informative :/

This seems to be the failure case, when `is_localtime` says no:

https://github.com/php/php-src/blob/php-8.2.11/ext/date/php_date.c#L3357-L3364
https://github.com/php/php-src/blob/php-8.2.11/ext/date/lib/timelib.h#L267

I don't understand the PHP source enough to figure out if/how this can actually happen.

Reply:
Maybe we could just assert() before? This way it shouldnt be a big deal if it can not happen under normal circumstances?